src/utils.rs

```diff
@@ -12,30 +12,22 @@
 //   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
 //
 use core::fmt;
-use std::collections::HashMap;
 use std::convert::TryFrom;
 use zenoh::prelude::KeyExpr;
-use zenoh::value::Value;
 use zenoh::Result as ZResult;
 use zenoh_keyexpr::OwnedKeyExpr;
 
-pub struct S3Value {
-    pub key: S3Key,
-    pub value: Value,
-    pub metadata: Option<HashMap<String, String>>,
-}
-
-pub struct S3Key {
-    pub prefix: Option<String>,
+pub struct S3Key<'a> {
+    pub prefix: &'a Option<String>,
     pub key: String,
 }
```

Comment:
Please change to this instead:
```suggestion
pub struct S3Key<'a> {
    pub prefix: Option<&'a str>,
    pub key: OwnedKeyExpr,
}
```
- This video covers in details why `Option<&T>`: https://www.youtube.com/watch?v=6c7pZYP_iIE
- An `OwnedKeyExpr` contains an `Arc<str>`, less expensive to clone than a `String`